### PR TITLE
Memtest example for Rocksdb

### PR DESCRIPTION
**Pull request by NikVolf**
this is half-test, half-example how to use Rocksdb

~~Not sure if it needs to get merged atm~~

### CODE REVIEW

File: kvdb-rocksdb/examples/memtest.rs [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+            buf[0..32].copy_from_slice(seed.as_ref());
+            let val = db.get(col, k.as_ref()).expect("Db fail").expect("Was put above");
+            buf[32..64].copy_from_slice(val.as_ref());
+
+            seed = keccak(&buf[..]);
+        }
+        keyvalues = KeyValueSeed::with_seed(seed);
+
+        if step % 10000 == 9999 {
+            println!("sys memory footprint: {} Mb", proc_memory_usage() / 1024);
```
Comment (dvdplm) [inline kvdb-rocksdb/examples/memtest.rs]:
```suggestion
            println!("Memory used as seen by the OS: {} Mb", proc_memory_usage() / 1024);
```

File: kvdb-rocksdb/examples/memtest.rs [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            println!("lib memory report: {} Mb", parity_util_mem::malloc_size(&db) / 1024 / 1024);
```
Comment (dvdplm) [inline kvdb-rocksdb/examples/memtest.rs]:
```suggestion
            println!("Memory used as seen by the process: {} Mb", parity_util_mem::malloc_size(&db) / 1024 / 1024);
```

File: kvdb-rocksdb/examples/memtest.rs [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+
+    let mut config = DatabaseConfig::with_columns(COLUMN_COUNT);
+
+    for c in 0..=COLUMN_COUNT {
+        config.memory_budget.insert(c, mb_per_col);
+    }
+    let dir = tempdir::TempDir::new("rocksdb-example").unwrap();
+
+    println!("Database is put in: {} (maybe check if it was deleted)", dir.path().to_string_lossy());
```
Comment (dvdplm) [inline kvdb-rocksdb/examples/memtest.rs]: I think tempdirs are deleted as soon as the process ends, so not sure what you mean by "maybe check if it was deleted"?

Comment (NikVolf) [reply kvdb-rocksdb/examples/memtest.rs]: Well it is not deleted when I ctrl-c currently 🤷‍♀️

File: kvdb-rocksdb/examples/memtest.rs [hunk 4]
```diff
@@ -0,0 +1,130 @@
+use ethereum_types::H256;
```
Comment (ordian) [inline kvdb-rocksdb/examples/memtest.rs]: What is our policy on license headers, do you need to include them in every single file?
I don't really care, just want it to be consistent.

Comment (dvdplm) [reply kvdb-rocksdb/examples/memtest.rs]: Yes, until further notice we should include them. I am 99% sure it's wholly unneeded, but until we know, they should be there.

@NikVolf can add pls?

File: kvdb-rocksdb/Cargo.toml [hunk 5]
```diff
@@ -31,3 +31,7 @@ ethereum-types = { path = "../ethereum-types" }
 kvdb-shared-tests = { path = "../kvdb-shared-tests", version = "0.2" }
 rand = "0.7.2"
 tempdir = "0.3.7"
+keccak-hash = { path = "../keccak-hash" }
+sysinfo = "0.9"
+ctrlc = "3.1.4"
+time = "0.1"
```
Comment (ordian) [inline kvdb-rocksdb/Cargo.toml]: why using old versions of `time` (latest is `0.2.9`) and `sysinfo` (latest is `0.11.7`)?

File: kvdb-rocksdb/examples/memtest.rs [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+// Parity Ethereum is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity Ethereum.  If not, see <http://www.gnu.org/licenses/>.
+
+// This program starts writing random data to the database with 100 (COLUMN_COUNT)
+// columns and never stops until interrapted.
```
Comment (NikVolf) [inline kvdb-rocksdb/examples/memtest.rs]:
```suggestion
// columns and never stops until interrupted.
```

File: kvdb-rocksdb/Cargo.toml [hunk 7]
```diff
@@ -31,3 +31,7 @@ ethereum-types = { path = "../ethereum-types" }
 kvdb-shared-tests = { path = "../kvdb-shared-tests", version = "0.2" }
 rand = "0.7.2"
 tempdir = "0.3.7"
+keccak-hash = { path = "../keccak-hash" }
+sysinfo = "0.11.7"
+ctrlc = "3.1.4"
+time = "0.2.9"
```
Comment (dvdplm) [inline kvdb-rocksdb/Cargo.toml]: I think the API changed and now CI's hating on us. :/

Comment (ordian) [reply kvdb-rocksdb/Cargo.toml]: Alternatively, we could use https://docs.rs/humantime/2.0.0/humantime/fn.format_rfc3339.html

Comment (NikVolf) [reply kvdb-rocksdb/Cargo.toml]: I have no idea why the made 0.2 version completely unusable, returned 0.1

**Pull request merged**